playback/src/audio_backend/rodio.rs

```diff
@@ -1,109 +1,165 @@
     rodio_sink: rodio::Sink,
-    // We have to keep hold of this object, or the Sink can't play...
-    #[allow(dead_code)]
-    stream: rodio::OutputStream,
+
+    // will produce a TryRecvError on the receiver side when it is dropped.
+    _close_tx: mpsc::SyncSender<Infallible>,
 }
 
-fn list_formats(ref device: &rodio::Device) {
-    let default_fmt = match device.default_output_config() {
-        Ok(fmt) => cpal::SupportedStreamConfig::from(fmt),
-        Err(e) => {
-            warn!("Error getting default rodio::Sink config: {}", e);
-            return;
+fn list_formats(device: &rodio::Device) {
+    match device.default_output_config() {
+        Ok(cfg) => {
+            debug!("  Default config:");
+            debug!("    {:?}", cfg);
         }
-    };
-    debug!("  Default config:");
-    debug!("    {:?}", default_fmt);
-
-    let mut output_configs = match device.supported_output_configs() {
-        Ok(f) => f.peekable(),
         Err(e) => {
-            warn!("Error getting supported rodio::Sink configs: {}", e);
-            return;
+            // Use loglevel debug, since even the output is only debug
+            debug!("Error getting default rodio::Sink config: {}", e);
         }
     };
 
-    if output_configs.peek().is_some() {
-        debug!("  Available configs:");
-        for format in output_configs {
-            debug!("    {:?}", format);
+    match device.supported_output_configs() {
+        Ok(mut cfgs) => {
+            if let Some(first) = cfgs.next() {
+                debug!("  Available configs:");
+                debug!("    {:?}", first);
+            } else {
+                return;
+            }
+
+            for cfg in cfgs {
+                debug!("    {:?}", cfg);
+            }
+        }
+        Err(e) => {
+            debug!("Error getting supported rodio::Sink configs: {}", e);
         }
     }
 }
 
-fn list_outputs() {
-    let default_device = get_default_device();
-    let default_device_name = default_device.name().expect("cannot get output name");
-    println!("Default Audio Device:\n  {}", default_device_name);
-    list_formats(&default_device);
-
-    println!("Other Available Audio Devices:");
-    for device in cpal::default_host()
-        .output_devices()
-        .expect("cannot get list of output devices")
-    {
-        let device_name = device.name().expect("cannot get output name");
-        if device_name != default_device_name {
-            println!("  {}", device_name);
-            list_formats(&device);
+fn list_outputs() -> Result<(), cpal::DevicesError> {
+    let mut default_device_name = None;
+
+    if let Some(default_device) = get_default_device() {
+        default_device_name = default_device.name().ok();
+        println!(
+            "Default Audio Device:\n  {}",
+            default_device_name.as_deref().unwrap_or("[unknown name]")
+        );
+
+        list_formats(&default_device);
+
+        println!("Other Available Audio Devices:");
```

Comment:
Was `println!` before, but would this not make more sense to be `info!`? Like in general have no `println!` in the lib because it can not be filtered?

Reply:
This is just executed in the case someone specifies "?" as device name and explicitly wishes to see the available devices.